core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java

```diff
@@ -147,6 +148,9 @@ private static void putReportedMetrics(IndexedSet<Metric> metricSet, String inst
         double oldVal = oldMetric == null ? 0.0 : oldMetric.getValue();
         Metric newMetric = new Metric(metric.getInstanceType(), metric.getHostname(),
             metric.getMetricType(), metric.getName(), oldVal + metric.getValue());
+        for (Map.Entry<String, String> tag : metric.getTags().entrySet()) {
+          newMetric.addTag(tag.getKey(), tag.getValue());
+        }
```

Comment:
So, if my understanding is correct, the extra tags contain values which the master will aggregate and display in the UI/Metrics report?

Reply:
For UFS read/write specifically, the master will aggregate all the values regardless of the tags (debatable whether or not we want to expose per UFS metrics).